internal/controller/apiextensions/composite/reconciler.go

```diff
@@ -757,3 +786,133 @@ func EnqueueForCompositionRevisionFunc(of resource.CompositeKind, list func(ctx
 		}
 	}
 }
+
+// handleFunctionResults will process the result returned by the composition functions.
+// This includes recording events for the xr and claim and setting updating status conditions
+// on the xr.
+func handleFunctionResults(r *Reconciler, log logging.Logger, cm *claim.Unstructured, xr *composite.Unstructured, res CompositionResult) {
```

Comment:
Info: Added `handleFunctionResults` function as it is needed in two separate places, depending on failure/non failure paths.